apis/tpu/v1alpha1/types.generated.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+// +kcc:proto=google.cloud.tpu.v2.AcceleratorConfig
+type AcceleratorConfig struct {
+	// Required. Type of TPU.
+	// +kcc:proto:field=google.cloud.tpu.v2.AcceleratorConfig.type
+	Type *string `json:"type,omitempty"`
+
+	// Required. Topology of TPU in chips.
+	// +kcc:proto:field=google.cloud.tpu.v2.AcceleratorConfig.topology
+	Topology *string `json:"topology,omitempty"`
+}
+
+// +kcc:proto=google.cloud.tpu.v2.AttachedDisk
+type AttachedDisk struct {
+	// Specifies the full path to an existing disk.
+	//  For example: "projects/my-project/zones/us-central1-c/disks/my-disk".
+	// +kcc:proto:field=google.cloud.tpu.v2.AttachedDisk.source_disk
+	SourceDisk *string `json:"sourceDisk,omitempty"`
+
+	// The mode in which to attach this disk.
+	//  If not specified, the default is READ_WRITE mode.
+	//  Only applicable to data_disks.
+	// +kcc:proto:field=google.cloud.tpu.v2.AttachedDisk.mode
+	Mode *string `json:"mode,omitempty"`
+}
+
+// +kcc:proto=google.cloud.tpu.v2.NetworkConfig
+type NetworkConfig struct {
+	// The name of the network for the TPU node. It must be a preexisting Google
+	//  Compute Engine network. If none is provided, "default" will be used.
+	// +kcc:proto:field=google.cloud.tpu.v2.NetworkConfig.network
+	Network *string `json:"network,omitempty"`
+
+	// The name of the subnetwork for the TPU node. It must be a preexisting
+	//  Google Compute Engine subnetwork. If none is provided, "default" will be
+	//  used.
+	// +kcc:proto:field=google.cloud.tpu.v2.NetworkConfig.subnetwork
+	Subnetwork *string `json:"subnetwork,omitempty"`
+
+	// Indicates that external IP addresses would be associated with the TPU
+	//  workers. If set to false, the specified subnetwork or network should have
+	//  Private Google Access enabled.
+	// +kcc:proto:field=google.cloud.tpu.v2.NetworkConfig.enable_external_ips
+	EnableExternalIps *bool `json:"enableExternalIps,omitempty"`
```

Comment:
nit: `Ips` --> `IPs`

Reply:
This is generated so I'll fix in a separate PR

Reply:
Scratch that, as I had to fix Network and Subnetwork in this struct anyway to be refs.  Added a linter in #3881